src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveGatherVectorFirstFaultingIndices.template

```diff
@@ -0,0 +1,633 @@
+        }
+
+        public void RunBasicScenario_Load()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_Load));
+
+            {Op1VectorType}<{Op1BaseType}> loadMask1 = Sve.CreateTrueMask{Op1BaseType}(SveMaskPattern.All);
+            {Op3VectorType}<{Op3BaseType}> loadMask3 = Sve.CreateTrueMask{Op3BaseType}(SveMaskPattern.All);
+
+            var result = {Isa}.{Method}(
+                {LoadIsa}.Load{Op1VectorType}(loadMask1, ({Op1BaseType}*)(_dataTable.inArray1Ptr)),
+                ({Op2BaseType}*)_dataTable.inArray2Ptr,
+                {LoadIsa}.Load{Op3VectorType}(loadMask3, ({Op3BaseType}*)(_dataTable.inArray3Ptr))
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_dataTable.inArray1Ptr, _dataTable.inArray2Ptr, _dataTable.inArray3Ptr, _dataTable.outArrayPtr);
+        }
+
+        public void RunBasicScenario_LoadFirstFaulting()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_LoadFirstFaulting));
+
+            {Op1VectorType}<{Op1BaseType}> loadMask1 = Sve.CreateTrueMask{Op1BaseType}(SveMaskPattern.All);
+            {Op3VectorType}<{Op3BaseType}> loadMask3 = Sve.CreateTrueMask{Op3BaseType}(SveMaskPattern.All);
+
+            var op1 = {LoadIsa}.Load{Op1VectorType}(loadMask1, ({Op1BaseType}*)(_dataTable.inArray1Ptr));
+            ref var op2Ref = ref (_dataTable.inBounded.Span.GetPinnableReference());
+            var op3 = {LoadIsa}.Load{Op3VectorType}(loadMask3, ({Op3BaseType}*)(_dataTable.inArray3Ptr));
+
+            // When testing first-faulting behavior, we need to make sure we can get the first element.
+            // So set the first active element of the index vector to 0.
+            for (var i = 0; i < Vector<{Op3BaseType}>.Count; i++)
+            {
+                // op1 is the mask for GatherVector.
+                if (op1[i] != 0)
+                {
+                    op3 = op3.WithElement<{Op3BaseType}>(i, 0);
+                    break;
+                }
+            }
+
+            Sve.SetFfr(Sve.CreateTrueMaskByte(SveMaskPattern.All));
+            var result = {Isa}.{Method}(
+                op1,
+                ({Op2BaseType}*)(Unsafe.AsPointer(ref op2Ref)),
+                op3
+            );
+            var faultResult = Sve.GetFfr{GetFfrType}();
+
+            ref var op3Ref = ref op3;
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateFirstFaultingResult(_dataTable.inArray1Ptr, ref op2Ref, _dataTable.inBounded.Span.Length, Unsafe.AsPointer(ref op3Ref), _dataTable.outArrayPtr, faultResult.As<{GetFfrType}, {RetBaseType}>());
+        }
+
+        public void RunBasicScenario_FalseMask()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunBasicScenario_FalseMask));
+
+            {Op1VectorType}<{Op1BaseType}> falseMask = Sve.CreateFalseMask{Op1BaseType}();
+
+            var result = {Isa}.{Method}(
```

Comment:
can we verify that the value of `Ffr` remains unchanged?

Reply:
What do you mean unchanged in this context? What would I be checking?

Reply:
Basically something like:

```c#
var beforeFfr = GetFfr();
... = GatherLoad()
var afterFfr = GetFfr();

assert(beforeFfr == afterFfr);
```